File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        res = ""
        for ch in s:
            if len(res) >= len(part)-1 and ch == part[-1] and res[len(res)-(len(part)-1):] == part[:len(part)-1]:
                res = res[:len(res)-(len(part)-1)]
            else:
                res+=ch
        return res

File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.java
class Solution {
    public String removeOccurrences(String s, String part) {
        StringBuilder res = new StringBuilder();
        int l = 0;
        while(l < s.length()){
            res.append(s.charAt(l));
            if(res.length() >= part.length() && res.toString().substring(res.length()-part.length(),res.length()).equals(part)){
                res.delete(res.length()-part.length(),res.length());
            }
            l++;
        }
        return res.toString();
    }
}